src_analysis/diff_utils.py

```diff
@@ -17,27 +17,47 @@
 import subprocess
 from typing import List
 
+from common import utils
+
+
+class DiffError(Exception):
+    """An error diffing commits."""
+
+
+def execute_git_diff(diff_args, repo=utils.ROOT_DIR):
+    """Adds |diff_args| to the command 'git diff' and executes the command in
+    |repo|. Returns a list of each line in the output."""
+    command = ['git', 'diff'] + diff_args
+    previous_working_directory = os.getcwd()
+    # Change directories instead of using "git -C" because "HEAD" can't be used
+    # with "git -C".
+    try:
+        if previous_working_directory != repo:
+            os.chdir(repo)
+        return subprocess.check_output(command).decode().splitlines()
```

Comment:
use cwd in subprocess.check_output instead of previous_working_directory.

Reply:
Good suggestion!